crates/loam-cli/src/commands/build/clients.rs

```diff
@@ -399,38 +406,75 @@ export default new Client.Client({{
         Ok(())
     }
 
+    fn resolve_line(re: &Regex, line: &str, shell: &str, flag: &str) -> Result<String, Error> {
+        let mut result = String::new();
+        let mut last_match = 0;
+        for cap in re.captures_iter(line) {
+            let whole_match = cap.get(0).unwrap();
+            result.push_str(&line[last_match..whole_match.start()]);
+            let cmd = &cap[1];
+            let output = Self::execute_subcommand(shell, flag, cmd)?;
+            result.push_str(&output);
+            last_match = whole_match.end();
+        }
+        result.push_str(&line[last_match..]);
+        Ok(result)
+    }
+
+    fn execute_subcommand(shell: &str, flag: &str, cmd: &str) -> Result<String, Error> {
+        match Command::new(shell).arg(flag).arg(cmd).output() {
+            Ok(output) => {
+                let stdout = String::from_utf8_lossy(&output.stdout).trim().to_string();
+                let stderr = String::from_utf8_lossy(&output.stderr).trim().to_string();
+
+                if output.status.success() {
+                    Ok(stdout)
+                } else {
+                    Err(Error::SubCommandExecutionFailure(cmd.to_string(), stderr))
+                }
+            }
+            Err(e) => Err(Error::SubCommandExecutionFailure(
+                cmd.to_string(),
+                e.to_string(),
+            )),
+        }
+    }
+
     async fn run_init_script(
         &self,
         name: &str,
         contract_id: &str,
         init_script: &str,
     ) -> Result<(), Error> {
+        let re = Regex::new(r"\$\((.*?)\)").expect("Invalid regex pattern");
+
+        let (shell, flag) = if cfg!(windows) {
+            ("cmd", "/C")
+        } else {
+            ("sh", "-c")
```

Comment:
Can we expected every unix machine to have `sh`?

Reply:
`sh` is part of the POSIX standard, which most Unix systems should comply with. I'm sure there might be some old machines that don't have it but I'm not sure we need to worry about that. 

see https://pubs.opengroup.org/onlinepubs/9699919799/
https://www.reddit.com/r/commandline/comments/vc8npd/posix_compliant_shells/

That being said, if the user wants to use a command that would not work in `sh`, I think it makes sense to add an option to specify your shell.